hazelcast/src/main/java/com/hazelcast/config/AutoDetectionConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Configuration for the Discovery Strategy Auto Detection.
+ * <p>
+ * When enabled, it will walk through all available discovery strategies and detect the correct one for the current runtime
+ * environment. For example, it will automatically use the {@code hazelcast-aws} plugin if run on an AWS instance.
+ */
+public class AutoDetectionConfig {
```

Comment:
`toString()` missing -> `JoinConfig.toString()` relies on it.

Reply:
Added